pkg/hns/endpoint_windows.go

```diff
@@ -40,6 +51,92 @@ func GetSandboxContainerID(containerID string, netNs string) string {
 	return containerID
 }
 
+func GenerateHnsEndpoint(epInfo *EndpointInfo, n *NetConf) *hcsshim.HNSEndpoint {
+	// run the IPAM plugin and get back the config to apply
+	hnsEndpoint, err := hcsshim.GetHNSEndpointByName(epInfo.EndpointName)
```

Comment:
This error is not being handled

Reply:
It seems like this err could still be missed. What if `hnsEndpoint == nil && err != nil`?

It looks like you're treating `err != nil` as the endpoint not existing. But it could also be a transient error, and you should handle that case.